### PR TITLE
37 a api drink sad path/edge case

### LINKED ISSUE #37: A API Drink - Sad Path/Edge Case
- [ ] Test for no Drink ID is received
- [ ] Test for an invalid ID requested
- [ ] Test for if a name is entered, not drink ID
- [ ] Test that a GraphQL error is returned if a field that does not exist for a drink is requested
- [ ] Test the return if a field is requested twice in the same query

### PR DESCRIPTION
**Pull request by aleish-m**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please describe): 
    - Sad path and edge case testing for API drink request

## Problem/feature

_What problem are you trying to solve?_

_Issue Number?_
Fixes #37 

## Checklist

- [x] Code compiles correctly
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: app/facades/cocktail_facade.rb [hunk 1]
```diff
@@ -36,18 +36,28 @@ def self.three_random
...
+    cocktail = CocktailService.get_cocktail_by_id(id)[:drinks]
+    if cocktail
+      cocktail = cocktail.first
+      ingredients = find_ingredients(cocktail)
+      ingredients.map! do |ingredient|
+        Ingredient.new(name: ingredient[:name], quantity: ingredient[:measurement])
       end
+      Drink.new(id: cocktail[:idDrink], name: cocktail[:strDrink], img_url: cocktail[:strDrinkThumb], steps: cocktail[:strInstructions], ingredients: ingredients)
+    else
+      raise ActiveRecord::RecordNotFound.new("Couldn't find Cocktail with 'id'=#{id}")
```
Comment (musselmanth) [inline app/facades/cocktail_facade.rb:35]: I like the repurposing of the Active::RecordNotFound error here because we already have the functionality for that error built into graphql

File: app/facades/cocktail_facade.rb [hunk 2]
```diff
@@ -20,34 +13,38 @@ def self.by_name(query)
...
-    ingredients = []
-    15.times do |i|
-      if cocktail["strIngredient#{i+1}".to_sym]
-        name = cocktail["strIngredient#{i+1}".to_sym]
-        measurement = cocktail["strMeasure#{i+1}".to_sym]
-        ingredients << { name: name, measurement: measurement }
+    cocktail = CocktailService.get_cocktail_by_id(id)[:drinks]
+    if cocktail
+      cocktail = cocktail.first
+      ingredients = find_ingredients(cocktail)
```
Comment (musselmanth) [inline app/facades/cocktail_facade.rb:29]: This was a much needed helper method. nice job

**Review by musselmanth — APPROVED** [review]
